docs/en/potentiometer.md

```diff
@@ -0,0 +1,146 @@
+potentiometer = PotentiometerHandler()
+keyboard.modules.append(potentiometer)
+```
+
+**2. Define the pins for each potentiometer: `pin_a` for the signal pin and `defname` for the name of the controlling defenition. If you want to invert the direction of the potentiometer, set the 3rd (optional) parameter `is_inverted` to `True`.**
+
+```python
+potentiometer.pins = (
+    (board."pin_a", "defname", "is_inverted")
+)
+```
+
+ Example:
+```python
+# Regular GPIO potentiometer
+potentiometer.pins = (
+    # regular direction potentiometer
+    (board.A0, potentiometer_1_handler),
+    # reversed direction potentiometer
+    (board.A1, potentiometer_2_handler, False),
+)
+```
+
+**3. Define the mapping of keys to be called.**
+
+here we convert the incoming base values into a value 0-127 for ease of use.
+
+This example is for a joystick mapped to WASD with a deadzone in the center. The exact deadzone values might vary depending on the potentiometers used.
+    
+    *Note: this uses `keyboard.add_key` and `keyboard.remove_key` which could be considered legacy.*
+
+```python
+def potentiometer_1_handler(state):
+    joy1 = int((state.position / 127) * 127)
+    if joy1 >= 0 and joy1 <= 56:
+        keyboard.add_key(KC.S)
+    if joy1 >= 65 and joy1 <= 127:
+        keyboard.add_key(KC.W)
+    if joy1 >= 57 and joy1 <= 64:
+        keyboard.remove_key(KC.S)
+        keyboard.remove_key(KC.W)
+
+def potentiometer_2_handler(state):
+    joy2 = int((state.position / 127) * 127)
+    if joy2 >= 0 and joy2 <= 58:
+        keyboard.add_key(KC.A)
+    if joy2 >= 67 and joy2 <= 127:
+        keyboard.add_key(KC.D)
+    if joy2 >= 59 and joy2 <= 66:
+        keyboard.remove_key(KC.A)
+        keyboard.remove_key(KC.D)
+```
+
+## Other examples
+
+**Computer volume**
+
+You can use a potentiometer to control the system volume easily. Here an example from [ZFR_KBD's RP2.65-F](https://github.com/KMKfw/kmk_firmware/blob/master/user_keymaps/ZFR_KBD/RP2.65-F.py)
+
+```python
+def set_sys_vol(state):
+    # convert to 0-100
+    new_pos = int((state.position / 127) * 64)
+    level = level_lut[new_pos]
+    # print(f"new vol level: {level}")
+    # print(f"last: {keyboard.last_level}")
+
+    # check if uninitialized
+    if keyboard.last_level == -1:
+        keyboard.last_level = level
+        return
+
+    level_diff = abs(keyboard.last_level - level)
+    if level_diff > 0:
+        # set volume to new level
+        # vol_direction = "unknown"
+        if level > keyboard.last_level:
+            # vol_direction = "up"
+            cmd = KC.VOLU
+        else:
+            # vol_direction = "down"
+            cmd = KC.VOLD
+
+        # print(f"Setting system volume {vol_direction} by {level_diff} to reach {level}")
+        for i in range(int(level_diff / level_inc_step)):
+            hid_report = keyboard._hid_helper.create_report([cmd])
+            hid_report.send()
+            hid_report.clear_all()
+            hid_report.send()
```

Comment:
i am not exactly sure why the hid_report was in there as this is a copied example from ZFR_KBD's RP2.65-F keymap that is included with the KMK repo. I wil have to look into how circuitpython can interface with the os if not by utilizing the hid_report. I am not that well versed in exactly this field so if you or someone else has a better example for this that would be great.

Reply:
I don't think we should put any kind of jank volume control examples into the "official" documentation, purely because of the third bullet point. The same "level" of volume on your keyboard will always produce different levels of volume on your computer. Terrible user experience.